src/main/java/edu/berkeley/cs186/database/index/BPlusTree.java

```diff
@@ -234,84 +235,100 @@ public Iterator<RecordId> scanGreaterEqual(DataBox key) {
         // TODO(proj4_integration): Update the following line
         LockUtil.ensureSufficientLockHeld(lockContext, LockType.NL);
 
-        // TODO(proj2): Return a BPlusTreeIterator.
+        // TODO(proj2): Return a BPlusTreeIterator. Done
 
-        return Collections.emptyIterator();
+        return new BPlusTreeIterator(root.getLeftmostLeaf(), key);
     }
 
     /**
      * Inserts a (key, rid) pair into a B+ tree. If the key already exists in
      * the B+ tree, then the pair is not inserted and an exception is raised.
-     *
-     *   DataBox key = new IntDataBox(42);
-     *   RecordId rid = new RecordId(42, (short) 42);
-     *   tree.put(key, rid); // Success :)
-     *   tree.put(key, rid); // BPlusTreeException :(
+     * <p>
+     * DataBox key = new IntDataBox(42);
+     * RecordId rid = new RecordId(42, (short) 42);
+     * tree.put(key, rid); // Success :)
+     * tree.put(key, rid); // BPlusTreeException :(
      */
     public void put(DataBox key, RecordId rid) {
         typecheck(key);
         // TODO(proj4_integration): Update the following line
         LockUtil.ensureSufficientLockHeld(lockContext, LockType.NL);
 
-        // TODO(proj2): implement
+        // TODO(proj2): implement Done
         // Note: You should NOT update the root variable directly.
         // Use the provided updateRoot() helper method to change
         // the tree's root if the old root splits.
 
-        return;
+        Optional<Pair<DataBox, Long>> promote = root.put(key, rid);
+        promote.ifPresent(n -> splitRoot(promote.get()));
```

Comment:
```suggestion
        promote.ifPresent(this::splitRoot);
```
이렇게 메서드를 직접 참조해서 쓸 수 있지 않을까요? promote가 optional이라 get()을 쓰면 Optional이 비어있을 때 예외를 던질 수도 있을거같아요

Reply:
좋네요!!

Reply:
수정하려고 했더니 인자를 받는 메서드라 다른 방법을 찾아야겠네요..

Reply:
인자가 들어가서 안될 줄 알았는데 알아서 들어가네요!  👍 👍 👍